giotto/mapper/visualization.py

```diff
@@ -295,153 +172,216 @@ def create_network_3d(graph, data, node_pos, node_color, columns_to_color=None,
-        select the columns to color by.
+    pipeline : :class:`MapperPipeline` object
+        Mapper pipeline to act on to data.
+
+    data : array-like of shape (n_samples, n_features)
+        Data used to generate the Mapper graph. Can be a pandas dataframe.
+
+    layout : None, str or callable, optional, default: ``'kamada-kawai'``
+        Layout algorithm for the graph. Can be any accepted value for the
+        ``layout`` parameter in the :meth:`layout` method of
+        :class:`igraph.Graph`. [1]_
+
+    layout_dim : int, default: ``2``
+        The number of dimensions for the layout. Can be 2 or 3.
+
+    color_variable : column index or name, or list of such, \
+        or ndarray/pandas dataframe of shape (n_samples, n_target_features), \
+        or (fit-)transformer object, or None, optional, default: ``None``
+        Specifies which quantity is to be used for node colouring. When it is
+        a numpy ndarray or pandas dataframe, it must have the same length as
+        `data` and is interpreted as a quantity of interest according to
+        which each node of the Mapper graph is to be coloured (see
+        :attr:`node_colors`). ``None`` is equivalent to passing `data`. If an
+        object implementing :meth:`transform` or :meth:`fit_transform`,
+        e.g. a scikit-learn estimator or pipeline, it is applied to `data`
+        to generate the quantity of interest. Otherwise, it must be a column
+        or subset of columns to be selected from `data`.
+
+    node_color_statistic : callable, or ndarray of shape (n_nodes,) or \
+        (n_nodes, 1), optional, default: ``numpy.mean``
+        Specifies how to determine the colors of each node. If a
+        numpy array, it must have the same length as the number of nodes in
+        the Mapper graph, and its values are used directly for node
+        coloring, ignoring `color_variable`. Otherwise, it must be a
+        callable object and is used to obtain a summary statistic within
+        each Mapper node of the quantity specified by :attr:`node_colors`.
+
+    color_by_columns_dropdown : bool, optional, default: ``True``
+        If ``True``, a dropdown widget is generated which allows the user to
+        colour Mapper nodes according to any column in `data`.
 
     plotly_kwargs : dict, optional, default: ``None``
         Keyword arguments to configure the Plotly Figure.
 
-    summary_stat : callable, default ``np.mean``
-        Summary statistic to apply to the elements in each node of the
-        topological graph.
-
     """
 
     # TODO could abstract away common patterns in get_cover_params_widgets and
     #  get_cluster_params_widgets
-
-    # TODO allow dimension to be passed as either 2 or 3 as an arg or kwarg
-
-    # clone input pipeline to catch scenarios where user selects invalid
-    # configuration of parameters and re-executes cell in Jupyter notebook
     pipe = clone(pipeline)
 
     def get_cover_params_widgets(param, value):
```

Comment:
Can we have the cover type here too? I.e. balanced vs uniform

Reply:
Good catch!